011-php-code-formatting.md

```diff
@@ -0,0 +1,94 @@
+
+However, once flagged in a pull request, to automatically fix these errors, developers can just run the `composer format` command in Homestead and then commit and push up their code, confident that the errors were addressed and the style checks should pass.
+
+## Drawbacks
+
+### Two-Worlds Problem
+
+In an ideal world, the PHP CS Fixer tool would run after Prettier and both be triggered by the pre-commit hook. However, based on our development environment setup using Homestead, we typically run `composer` commands within the vagrant virtual machine and `npm` commands outside the vagrant box (watching and compiling assets with via `npm` tool has proven to run rather slow on the VM and thus we typically run them within our local environment).
+
+Due to this setup, one drawback with the PHPCS Fixer tool is that the tool cannot be triggered as a pre-commit hook to run when Prettier is run.
+
+This [post](https://sebastiandedeyne.com/running-php-cs-fixer-on-every-commit-with-husky-and-lint-staged/) discusses setting up both Prettier and PHP CS Fixer to run using the [`lint-staged`](https://github.com/okonet/lint-staged) NPM package, which allows configuring use of code quality tools alongside Prettier. However, I think it only works because they are both running within the same environment (and not split between a VM and local computer).
+
+While it is unfortunate it is not a deal breaker and as described in the "PHP Code Standards Fixer" section above, the checks will happen via the CI runs and devs can fix errors locally by running `composer format`, commit and push them up.
+
+### Contradictions
+
+Another potential drawback is the possibility of contradicting fixes. Where depending on the order of operations, a style fix could keep getting edited back and forth between Prettier and PHP CS Fixer.
+
+This is theoretical, since I have not encountered this, but I think it will be something to look out for initially if we move forward with this RFC.
+
+Luckily, both Prettier and the PHP CS Fixer rulesets adhere to the same PSRs so I am hopeful that this will not really be an issue.
+
+For a quick visual comparison, here is the same file edited separately by the two tools:
+
+#### Prettier Changes
+
+You can see in these [Prettier file changes](https://github.com/DoSomething/rogue/pull/1094/files#diff-5c9547d315e53dd95227ca9984b56d06), it edited the line-lengths of the code.
+
+#### PHP CS Fixer Changes
+
+You can see in these [PHP CS Fixer file changes](https://github.com/DoSomething/rogue/pull/1092/files#diff-5c9547d315e53dd95227ca9984b56d06), it sorted the class imports to be alphabetical and removed the spacing around string concatenation.
```

Comment:
I think the string concatenation rule is a good example of something we'll want to let Prettier handle! (Since it seems more like a "formatting" rule & I don't believe we can customize [Prettier's behavior]([let Prettier handle](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8M+VBZhqYAGpIlWfCACsAOj4gWMCG3iyVAYXQBbH2wAFAAkA-BdBQxAOVwBjKABlGABPNWswbAgAWgAmJxBoAEdBeGgA12V-PTkFEGIRCBgytgq2KoMAXxYoCngYdBQ0eCxJUDZsGDaRTtrMMEEJvBAAHgB+YzAAHUwtqBjqdAACAHIdcZUjuyPxiHO7QTAwJAAKI5isGLYYI4BKAG4QdpAA)) on this one & wouldn't want the two tools to conflict.)

Reply:
Ah that's a good point. When I initially reviewed the results of having run Prettier on the codebase it seemed like the concatenation rule was something it didn't address (assuming it would remove spaces 😅)... I hadn't realized it didn't touch them BECAUSE it considered that correct! haha. But good to know my theory on "contradicting" rules was right!

I think my planned approach, if approved, will be to first merge in the Prettier PR and establish what Prettier normally fixes. Knowing that I can customize a ruleset for PHP CS Fixer to add to `weerd/php-style` that doesn't conflict called `prettier-laravel` or something.

Reply:
That makes sense to me! 👍